### ISSUE TITLE
Loss function is not LMCL

### ISSUE THREAD

**Issue by petrLorenc** [thuiar/DeepUnkID#5]
How is this loss function:

```
def large_margin_cosine_loss(y_true, y_pred, scale=30, margin=0.35):
    y_pred = y_true * (y_pred - margin) + (1 - y_true) * y_pred
    y_pred *= scale
    return K.categorical_crossentropy(y_true, y_pred, from_logits=True)
```
related to 2.2 Large Margin Cosine Loss (LMCL) in the reference [paper](https://www.aclweb.org/anthology/P19-1548.pdf) 

It seems totally different, right?

**Comment by petrLorenc** [comment]
According to the original idea of the paper, the softmax layer should be applied after the margin/scale. In this case, it was applied beforehand so it seems not the same.
